docs/command-reference.md

```diff
@@ -0,0 +1,632 @@
+*   **Process control**
+    *   `-e, --env <key=value>`: set environment variables
+    *   `--env-file <file>`: read environment variables from a file
+    *   `-i, --interactive`: keep STDIN open for interactive processes
+    *   `-t, --tty`: allocate a pseudo-TTY
+    *   `-u, --user <user>`: run as the specified user name or UID
+    *   `--uid <uid>`: run as the specified numeric user ID
+    *   `--gid <gid>`: run as the specified numeric group ID
+    *   `--cwd <path>`: set the working directory inside the container
+*   **Container management**
+    *   `--name <name>`: assign a name to the container
+    *   `-d, --detach`: run the container in the background (daemon mode)
+    *   `--rm`: automatically remove the container when it exits
+    *   `--cidfile <file>`: write the container ID to a file
+*   **Registry**
+    *   `--scheme <scheme>`: registry scheme (`auto`, `http`, or `https-`)
+*   **Progress**
+    *   `--disable-progress-updates`: disable progress display for image pulls and pushes
+*   **Global**
+    *   `--debug`: enable debug logging
+    *   `-h, --help`: show help
+
+**Examples**
+
+```bash
+# run a container and attach an interactive shell
+container run -it ubuntu:latest /bin/bash
+
+# run a background web server
+container run -d --name web -p 8080:80 nginx:latest
+
+# set environment variables and limit resources
+container run -e NODE_ENV=production --cpus 2 --memory 1G node:18
+```
+
+### `container build`
+
+Builds an OCI image from a local build context. It reads a Dockerfile (default `Dockerfile`) and produces an image tagged with `-t` option. The build runs in isolation using BuildKit, and resource limits may be set for the build process itself.
+
+**Usage**
+
+```bash
+container build [OPTIONS] PATH
+```
+
+**Options**
+
+*   **Resource management**
+    *   `-c, --cpus <number>`: CPUs to allocate to the build process (default 2)
+    *   `-m, --memory <size>`: memory for the build process (default 2048MB)
+*   **Build configuration**
+    *   `--build-arg <key=value>`: build-time variables passed to the Dockerfile
+    *   `-f, --file <path>`: path to the Dockerfile (default `Dockerfile`)
+    *   `-l, --label <key=value>`: add metadata labels to the image
+    *   `--no-cache`: disable cache usage
+    *   `-o, --output <config>`: specify build output (default `type=oci`)
+    *   `--arch <arch>`: target architecture (default `arm64`)
+    *   `--os <os>`: target operating system (default `linux`)
+    *   `--progress <type>`: progress output mode: `auto`, `plain`, or `tty`
+    *   `--vsock-port <port>`: vsock port used by BuildKit (default 8088)
+    *   `-t, --tag <name>`: set image name and tag
+    *   `--target <stage>`: set the target stage for multi-stage builds
+    *   `-q, --quiet`: suppress build output
+*   **Hidden options**
+    *   `--cache-in <config>`: configure build cache imports
+    *   `--cache-out <config>`: configure build cache exports
```

Comment:
I don't think we should include these if they're hidden and aren't supported yet